src/libraries/System.Resources.Extensions/tests/BinaryResourceWriterUnitTest.cs

```diff
@@ -500,6 +501,50 @@ public static void EmbeddedResourcesAreUpToDate()
             }
         }
 
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsBinaryFormatterSupported))]
+        /// <summary>
+        /// This test has multiple threads simultaneously loop over the keys of a moderately-sized resx using
+        /// <see cref="ResourceManager"/> and call <see cref="ResourceManager.GetString(string)"/> for each key.
+        /// This has historically been prone to thread-safety bugs because of the shared cache state and internal
+        /// method calls between RuntimeResourceSet and <see cref="DeserializingResourceReader"/>.
+        ///
+        /// Running with <paramref name="useEnumeratorEntry"/> TRUE replicates https://github.com/dotnet/runtime/issues/74868,
+        /// while running with FALSE replicates the error from https://github.com/dotnet/runtime/issues/74052.
+        /// </summary>
+        /// <param name="useEnumeratorEntry">
+        /// Whether to use <see cref="IDictionaryEnumerator.Entry"/> vs. <see cref="IDictionaryEnumerator.Key"/> when enumerating;
+        /// these follow fairly different code paths.
+        /// </param>]
+        [InlineData(false)]
+        [InlineData(true)]
+        public static void TestResourceManagerIsSafeForConcurrentAccessAndEnumeration(bool useEnumeratorEntry)
+        {
+            ResourceManager manager = new(
+                typeof(TestData).FullName,
+                typeof(TestData).Assembly,
+                typeof(DeserializingResourceReader).Assembly.GetType("System.Resources.Extensions.RuntimeResourceSet", throwOnError: true));
+
+            const int Threads = 10;
+            using Barrier barrier = new(Threads);
+            Task task = Task.WhenAll(Enumerable.Range(0, Threads).Select(_ => Task.Run(WaitForBarrierThenEnumerateResources)));
```

Comment:
We should probably avoid depending on the thread pool here, or else with the concurrency in unit tests these work items could be delayed arbitrarily long.
```suggestion
            Task task = Task.WhenAll(Enumerable.Range(0, Threads).Select(_ => Task.Factory.StartNew(WaitForBarrierThenEnumerateResources, CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default)));
```

Reply:
Changed to LongRunning task